Articles/ShortArticles/CodingConvensions.md

```diff
@@ -0,0 +1,109 @@
+
+While coding conventions for formatting, naming and other general guidelines may not impact the behavior of the software at all, they can have a dramatic influence on the readability and maintainability of the code, especially for expert programmers.
+In *Code Complete: 2nd Edition*<sup>[cc2nd04]</sup>, McConnell cites a study and notes:
+
+> "The importance to comprehension and memory of structuring one’s environment in a familiar way
+has led some researchers to hypothesize that layout might harm an expert’s ability to read a program
+if the layout is different from the scheme the expert uses (Shell 1981, Soloway and Ehrlich 1984)"
+[7, Section 31.1]. This implies that working with an unfamiliar style might handicap expert coders
+more than beginner and intermediate coders.
+
+Having a project, organization, or even an entire company adopt consistent coding conventions is useful to:
+
+* **increase productivity in the initial development** (by removing arbitrary choices and focusing effort on essentials),
+* **foster shared ownership of the codebase** (by using the same style one can't tell who wrote the code initially and therefore it does not imply ownership),
+* **streamline and improve peer code reviews** (by avoiding wasted time arguing about arbitrary choices and allowing tools to catch violations of the coding conventions so the reviewer does not need to bother with them),
+* **improve the understandably and maintainability of the codebase** (by using a coding style that is very familiar and consistent), and
+* **provide targets for refactorings** (by helping to highlight code problems that violate the accepted conventions).
+
+There are some excellent, widely accepted, and largely language-agnostic, code formatting and naming guidelines given in Chapter 31 "Layout and Style" of the book *Code Complete: 2nd Edition* by Steve McConnell and in Chapters 1 through 5 of the book *Clean Code*<sup>[cc08]</sup> by Robert Martin.
+Other chapters in *Code Complete: 2nd Edition* and *Clean Code* also cover many of the other general coding guidelines that are applicable to almost every programming language.
+These two authors are some of the most respected in the software industry whose guidance has held up over time and they give very consistent advice in this area.
+(But not that one should look to more recent references for language-specific guidelines instead of those given in *Code Complete: 2nd Edition*.)
+These guidelines apply to every procedural programming language (or the parts of program that are programmed using procedural approaches) and therefore apply to object-oriented programs as well.
+Other options exist in the software development community but the views expressed by these two authors seem to be fairly widely accepted.
+
+The different categories of conventions have different consequences, are often handled differently, and are more or less amiable to automated tools.
```

Comment:
I suggest: amiable --> amenable

Reply:
Good catch @bernhold!  Fixed in commit https://github.com/betterscientificsoftware/bssw.io/pull/1076/commits/3cf5cd254741ac40c0ba12553e77e877cb5c44af.